velox/functions/FunctionRegistry.cpp

```diff
@@ -160,4 +163,63 @@ resolveVectorFunctionWithMetadata(
   return exec::resolveVectorFunctionWithMetadata(functionName, argTypes);
 }
 
+const std::vector<std::string> getScalarNames(bool sortResult) {
+  // Do not print "internal" functions.
+  static const std::unordered_set<std::string> kBlockList = {
+      "row_constructor", "in", "is_null"};
+
+  auto functions = getFunctionSignatures();
+  std::vector<std::string> names;
+  names.reserve(functions.size());
+  exec::aggregateFunctions().withRLock([&](const auto& aggregateFunctions) {
+    for (const auto& func : functions) {
+      const auto& name = func.first;
+      if (!functions::isCompanionFunctionName(name, aggregateFunctions) &&
+          kBlockList.count(name) == 0) {
+        names.emplace_back(name);
+      }
+    }
+  });
+  if (sortResult) {
+    std::sort(names.begin(), names.end());
+  }
+  return names;
+}
+
+const std::vector<std::string> getAggregateNames(bool sortResult) {
+  std::vector<std::string> names;
+  names.reserve(exec::aggregateFunctions()->size());
```

Comment:
You probably need to move this to line 193 before executing the for loop (current build error). The lambda is only executed once under the r-lock so this should be ok.
And you probably want to use emplace_back in 195 to be more consistent - this is used in the other two functions. Either way a constructor is called - copy constructor or standard constructor (and then moved).

Reply:
Thanks @czentgr, updated accordingly.